### PR TITLE
refactor(prometheusrule): working prometheusrules

### PR DESCRIPTION
**Pull request by Hy3n4**

### CODE REVIEW

File: internal/controller/osko/mimirrule_controller.go [hunk 1]
```diff
@@ -203,6 +214,15 @@ func (r *MimirRuleReconciler) newMimirClient(ds *openslov1.Datasource) error {
...
 }
 
+//func (r *MimirRuleReconciler) getMimirRuleGroupAPI(log logr.Logger, name string) (*rwrulefmt.RuleGroup, error) {
+//	mimirRule, err := r.MimirClient.GetRuleGroup(context.Background(), mimirRuleNamespace, name)
+//	if err != nil {
+//		log.Error(err, "Failed to get rule group")
+//		return nil, err
+//	}
+//	return mimirRule, nil
+//}
```
Comment (fourstepper) [inline internal/controller/osko/mimirrule_controller.go:224]: we can probably delete this..

File: internal/helpers/prometheus_helper.go [hunk 2]
```diff
@@ -1,19 +1,278 @@
...
+
+	// We build the string by iterating over the sorted keys.
+	pairs := make([]string, len(labels))
+	for i, k := range keys {
+		pairs[i] = fmt.Sprintf("%s=\"%s\"", k, labels[k])
+	}
+
+	// Join the key-value pairs with commas and return the result.
+	return strings.Join(pairs, ", ")
+}
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:73]: smart, I like

File: internal/helpers/prometheus_helper.go [hunk 3]
```diff
@@ -1,19 +1,278 @@
...
+	if len(mrs.Slo.Spec.TimeWindow) > 0 && mrs.Slo.Spec.TimeWindow[0].Duration != "" {
+		window = string(mrs.Slo.Spec.TimeWindow[0].Duration)
+	}
+	rule.Labels = map[string]string{
+		"sli_name": mrs.Sli.Name,
+		"slo_name": mrs.Slo.Name,
+		"service":  mrs.Slo.Spec.Service,
+		"window":   window,
+	}
+}
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:273]: this function is unused now?

File: internal/helpers/prometheus_helper.go [hunk 4]
```diff
@@ -1,19 +1,278 @@
...
+
+func CreatePrometheusRule2(slo *openslov1.SLO, sli *openslov1.SLI) (*monitoringv1.PrometheusRule, error) {
```
Comment (fourstepper) [inline internal/helpers/prometheus_helper.go:275]: let's delete this whole func?

File: internal/utils/common_utils.go [hunk 5]
```diff
@@ -217,17 +217,25 @@ func getRelevantRule(b BudgetRule, goodRuleSpec, sloIndicatorSpec string, log lo
...
+//	measurement := monitoringv1.Rule{}
+//	measurement.Record = fmt.Sprintf("%s_%s", RecordPrefix, TypeMeasurement)
+//	exprFormat := "%s_%s{%s} / %s_%s{%s}"
+//	measurement.Expr = intstr.Parse(fmt.Sprintf(exprFormat,
+//		RecordPrefix, gbRule.Record, gbRule.MetricLabelCompiler.NewMetricLabelCompiler(nil, ""),
+//		RecordPrefix, totalRuleConfig.Record, totalRuleConfig.MetricLabelCompiler.NewMetricLabelCompiler(nil, ""),
+//	))
+//	return measurement
+//}
+
```
Comment (fourstepper) [inline internal/utils/common_utils.go:238]: I would delete this as well

**Review by fourstepper — CHANGES_REQUESTED** [review]

File: internal/controller/osko/mimirrule_controller.go (hunk 1 above)
Comment (Hy3n4) [reply internal/controller/osko/mimirrule_controller.go:224]: this is here for the future, but yes, it's not a gamechanger

**Review by fourstepper — APPROVED** [review]
LGTM

**Pull request merged**